<a href="https://colab.research.google.com/github/Untick/InspectrumClinic_RS_gr2/blob/Chervotkin-Evgeny-folder/Week_2_17_05_2023_24_05_2023.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Стажировка





# Задача: Исследовать датасет

*Для этого нужно подключить библиотеку Numpy.



In [2]:
# Загрузка модуля pandas
import pandas as pd

Затем нужно создать датасет. Для этого нужно загрузить датасет в Colab.

In [41]:
# Загрузка файла из облака в colab
import gdown
gdown.download('https://storage.yandexcloud.net/terratraineeship/23_InspectrumClinic_RS/datasets/%D0%B4%D0%B0%D1%82%D0%B0%D1%81%D0%B5%D1%82%20%D0%BF%D1%80%D0%BE%D1%84%D0%BF%D0%B0%D1%82%D0%BE%D0%BB%D0%BE%D0%B3%20%D1%87%D0%B0%D1%81%D1%82%D1%8C.xlsx', None, quiet=True)

'%D0%B4%D0%B0%D1%82%D0%B0%D1%81%D0%B5%D1%82%20%D0%BF%D1%80%D0%BE%D1%84%D0%BF%D0%B0%D1%82%D0%BE%D0%BB%D0%BE%D0%B3%20%D1%87%D0%B0%D1%81%D1%82%D1%8C.xlsx'

Создадаем переменную data, в которую с помощью функции pd.read_excel() загрузим данные из файла profpatolog.xlsx:

In [42]:
data = pd.read_excel('profpatolog.xlsx')


Смотрим что из себя представляет датасет.

In [8]:
data

ДокументПрохождениеМедосмотра  \
0     7f41ca7e-7dac-11ec-80fc-0cc47aab8067   
1     2ee1f9ac-83dd-11ec-80fc-0cc47aab8067   
2     9cbb191e-8e24-11ec-80fe-0cc47aab8067   
3     b8a33910-8eec-11ec-80fe-0cc47aab8067   
4     0bb0d181-938c-11ec-80fe-0cc47aab8067   
...                                    ...   
1977  137af572-70a5-11ed-8109-0cc47aab8067   
1978  26310d7e-70a5-11ed-8109-0cc47aab8067   
1979  b2602553-705c-11ed-8109-0cc47aab8067   
1980  d0358b83-7065-11ed-8109-0cc47aab8067   
1981  a15e278b-71f5-11ed-810a-0cc47aab8067   

                                    Клиент КлиентДатаРождения КлиентПол  \
0     c91e3407-b81f-11e9-80cf-0cc47aab8067         1976-10-24   Женский   
1     1a1a5fea-8b7f-11eb-80e8-0cc47aab8067         1987-11-18   Мужской   
2     7fe324e0-fa51-11eb-80f6-0cc47aab8067         1985-09-08   Мужской   
3     98f9d761-8d65-11ec-80fe-0cc47aab8067         1988-04-25   Мужской   
4     009aac7d-7bcf-11eb-80e8-0cc47aab8067         1984-04-17   Мужской   
...                                    ...                ...       ...   
1977  41a7c665-7098-11ed-8109-0cc47aab8067         1972-11-12   Мужской   
1978  41a7c663-7098-11ed-8109-0cc47aab8067         1986-07-09   Мужской   
1979  0dc52156-f1a9-11e9-80d1-0cc47aab8067         1994-05-12   Мужской   
1980  6261a674-07c8-11e8-80c6-0cc47aab8067         1972-04-17   Женский   
1981  920293ea-32c2-11eb-80e3-0cc47aab8067         1971-09-14   Мужской   

     КлиентПолПредставление ДатаЗавершенияМедосмотра  ТипМедосмотра  \
0                   Женский               2022-01-25  Периодический   
1                   Мужской               2022-02-02  Периодический   
2                   Мужской               2022-02-15  Периодический   
3                   Мужской               2022-02-16  Периодический   
4                   Мужской               2022-02-22  Периодический   
...                     ...                      ...            ...   
1977                Мужской               2022-11-30  Периодический   
1978                Мужской               2022-11-30  Периодический   
1979                Мужской               2022-11-30  Периодический   
1980                Женский               2022-11-30  Периодический   
1981                Мужской               2022-12-02  Периодический   

     ТипМедосмотраПредставление  ПсихОсвидетельствование  \
0                 Периодический                    False   
1                 Периодический                    False   
2                 Периодический                    False   
3                 Периодический                    False   
4                 Периодический                    False   
...                         ...                      ...   
1977              Периодический                    False   
1978              Периодический                    False   
1979              Периодический                    False   
1980              Периодический                    False   
1981              Периодический                    False   

                                Контрагент  ...  \
0     23c173a7-30d9-11e7-80d8-708bcd7f83cd  ...   
1     a6d559b0-7835-11eb-80e8-0cc47aab8067  ...   
2     a6d559b0-7835-11eb-80e8-0cc47aab8067  ...   
3     a6d559b0-7835-11eb-80e8-0cc47aab8067  ...   
4     a6d559b0-7835-11eb-80e8-0cc47aab8067  ...   
...                                    ...  ...   
1977  a9f70e4c-707f-11ed-8109-0cc47aab8067  ...   
1978  a9f70e4c-707f-11ed-8109-0cc47aab8067  ...   
1979  e689a382-17b2-11e9-80ce-0cc47aab8067  ...   
1980  21f2a93d-68f9-11e6-82dc-d897ba0a5c60  ...   
1981  4826f33a-06b9-11ea-80d3-0cc47aab8067  ...   

     Оториноларингология3_Аудиометрия1_Значение  \
0                                           NaN   
1                                           NaN   
2                                           NaN   
3                                           NaN   
4                                           NaN   
...                                         ...   
1977           

Выведем первые 5 строк из таблицы.

In [9]:
data.head()

ДокументПрохождениеМедосмотра                                Клиент  \
0  7f41ca7e-7dac-11ec-80fc-0cc47aab8067  c91e3407-b81f-11e9-80cf-0cc47aab8067   
1  2ee1f9ac-83dd-11ec-80fc-0cc47aab8067  1a1a5fea-8b7f-11eb-80e8-0cc47aab8067   
2  9cbb191e-8e24-11ec-80fe-0cc47aab8067  7fe324e0-fa51-11eb-80f6-0cc47aab8067   
3  b8a33910-8eec-11ec-80fe-0cc47aab8067  98f9d761-8d65-11ec-80fe-0cc47aab8067   
4  0bb0d181-938c-11ec-80fe-0cc47aab8067  009aac7d-7bcf-11eb-80e8-0cc47aab8067   

  КлиентДатаРождения КлиентПол КлиентПолПредставление  \
0         1976-10-24   Женский                Женский   
1         1987-11-18   Мужской                Мужской   
2         1985-09-08   Мужской                Мужской   
3         1988-04-25   Мужской                Мужской   
4         1984-04-17   Мужской                Мужской   

  ДатаЗавершенияМедосмотра  ТипМедосмотра ТипМедосмотраПредставление  \
0               2022-01-25  Периодический              Периодический   
1               2022-02-02  Периодический              Периодический   
2               2022-02-15  Периодический              Периодический   
3               2022-02-16  Периодический              Периодический   
4               2022-02-22  Периодический              Периодический   

   ПсихОсвидетельствование                            Контрагент  ...  \
0                    False  23c173a7-30d9-11e7-80d8-708bcd7f83cd  ...   
1                    False  a6d559b0-7835-11eb-80e8-0cc47aab8067  ...   
2                    False  a6d559b0-7835-11eb-80e8-0cc47aab8067  ...   
3                    False  a6d559b0-7835-11eb-80e8-0cc47aab8067  ...   
4                    False  a6d559b0-7835-11eb-80e8-0cc47aab8067  ...   

  Оториноларингология3_Аудиометрия1_Значение  \
0                                        NaN   
1                                        NaN   
2                                        NaN   
3                                        NaN   
4                                        NaN   

  Оториноларингология3_Аудиометрия1_ЗначениеПредставление  \
0                                                NaN        
1                                                NaN        
2                                                NaN        
3                                                NaN        
4                                                NaN        

  Оториноларингология1_ЖалобыДополнение_Значение  \
0                                            NaN   
1                                            NaN   
2                                            NaN   
3                                            NaN   
4                                            NaN   

  Оториноларингология1_ЖалобыДополнение_ЗначениеПредставление  \
0                                                NaN            
1                                                NaN            
2                                                NaN            
3                                                NaN            
4                                                NaN            

  Стоматология2_ВП2_Значение Стоматология2_ВП2_ЗначениеПредставление  \
0                        NaN                                     NaN   
1                        NaN                                     NaN   
2                        NaN                                     NaN   
3                        NaN                                     NaN   
4                        NaN                                     NaN   

  Терапия2_ЗаключениеФОГК_Значение  \
0                              NaN   
1                              NaN   
2                              NaN   
3                              NaN   
4                              NaN   

  Терапия2_ЗаключениеФОГК_ЗначениеПредставление  \
0                                           NaN   
1                                           NaN   
2                                           NaN   
3                                           NaN   
4                              

Крайние 5 строк.

In [10]:
data.tail()

ДокументПрохождениеМедосмотра  \
1977  137af572-70a5-11ed-8109-0cc47aab8067   
1978  26310d7e-70a5-11ed-8109-0cc47aab8067   
1979  b2602553-705c-11ed-8109-0cc47aab8067   
1980  d0358b83-7065-11ed-8109-0cc47aab8067   
1981  a15e278b-71f5-11ed-810a-0cc47aab8067   

                                    Клиент КлиентДатаРождения КлиентПол  \
1977  41a7c665-7098-11ed-8109-0cc47aab8067         1972-11-12   Мужской   
1978  41a7c663-7098-11ed-8109-0cc47aab8067         1986-07-09   Мужской   
1979  0dc52156-f1a9-11e9-80d1-0cc47aab8067         1994-05-12   Мужской   
1980  6261a674-07c8-11e8-80c6-0cc47aab8067         1972-04-17   Женский   
1981  920293ea-32c2-11eb-80e3-0cc47aab8067         1971-09-14   Мужской   

     КлиентПолПредставление ДатаЗавершенияМедосмотра  ТипМедосмотра  \
1977                Мужской               2022-11-30  Периодический   
1978                Мужской               2022-11-30  Периодический   
1979                Мужской               2022-11-30  Периодический   
1980                Женский               2022-11-30  Периодический   
1981                Мужской               2022-12-02  Периодический   

     ТипМедосмотраПредставление  ПсихОсвидетельствование  \
1977              Периодический                    False   
1978              Периодический                    False   
1979              Периодический                    False   
1980              Периодический                    False   
1981              Периодический                    False   

                                Контрагент  ...  \
1977  a9f70e4c-707f-11ed-8109-0cc47aab8067  ...   
1978  a9f70e4c-707f-11ed-8109-0cc47aab8067  ...   
1979  e689a382-17b2-11e9-80ce-0cc47aab8067  ...   
1980  21f2a93d-68f9-11e6-82dc-d897ba0a5c60  ...   
1981  4826f33a-06b9-11ea-80d3-0cc47aab8067  ...   

     Оториноларингология3_Аудиометрия1_Значение  \
1977                                        NaN   
1978                                        NaN   
1979                                        NaN   
1980                                        NaN   
1981                                        NaN   

     Оториноларингология3_Аудиометрия1_ЗначениеПредставление  \
1977                                                NaN        
1978                                                NaN        
1979                                                NaN        
1980                                                NaN        
1981                                                NaN        

     Оториноларингология1_ЖалобыДополнение_Значение  \
1977                                            NaN   
1978                                            NaN   
1979                                            NaN   
1980                                            NaN   
1981                                            NaN   

     Оториноларингология1_ЖалобыДополнение_ЗначениеПредставление  \
1977                                                NaN            
1978                                                NaN            
1979                                                NaN            
1980                                                NaN            
1981                                                NaN            

     Стоматология2_ВП2_Значение Стоматология2_ВП2_ЗначениеПредставление  \
1977                        NaN                                     NaN   
1978                        NaN                                     NaN   
1979                        NaN                                     NaN   
1980                        NaN                                     NaN   
1981                        NaN                                     NaN   

     Терапия2_ЗаключениеФОГК_Значение  \
1977                              NaN   
1978                              NaN   
1979                              NaN   
1980                              NaN   
1981                              NaN   

     Терапия2_ЗаключениеФОГК_ЗначениеПредставление  \
1977  

Первично датасет готов к обработке.

# Анализ датасета

Проанализируем данные для нейронной сети.

In [11]:
data.shape

(1982, 4587)

Имеет 1982 строки и 4587 ячеек .

In [12]:
# Сколько элементов в таблице
data.size 

9091434

In [13]:
data.count()

ДокументПрохождениеМедосмотра                                             1982
Клиент                                                                    1982
КлиентДатаРождения                                                        1982
КлиентПол                                                                 1982
КлиентПолПредставление                                                    1982
                                                                          ... 
Стоматология2_ВП2_ЗначениеПредставление                                      1
Терапия2_ЗаключениеФОГК_Значение                                             1
Терапия2_ЗаключениеФОГК_ЗначениеПредставление                                1
Офтальмология1_ДвигательныйАппаратГлазДополнение_Значение                    1
Офтальмология1_ДвигательныйАппаратГлазДополнение_ЗначениеПредставление       1
Length: 4587, dtype: int64

In [14]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1982 entries, 0 to 1981
Columns: 4587 entries, ДокументПрохождениеМедосмотра to Офтальмология1_ДвигательныйАппаратГлазДополнение_ЗначениеПредставление
dtypes: bool(1), datetime64[ns](43), float64(1253), object(3290)
memory usage: 69.3+ MB


DataFrame состоит из 1982 записей (строк) и 4587 столбцов.

Индекс (индексная колонка) имеет тип RangeIndex и содержит значения от 0 до 1981.
В DataFrame присутствуют различные типы данных:
1 cтолбец имеет тип bool.

43 столбца имеют тип datetime64[ns], что указывает на наличие дат и времени.

1253 столбца имеют тип float64, что указывает на наличие числовых значений с плавающей запятой.

3290 столбцов имеют тип object, что обычно означает наличие строковых данных или смешанных типов данных.

Общий объем памяти, используемый для хранения DataFrame, составляет примерно 69,3 мегабайта.

In [17]:
#Смотрим есть ли пустые ячейки
data.isna().head()

ДокументПрохождениеМедосмотра  Клиент  КлиентДатаРождения  КлиентПол  \
0                          False   False               False      False   
1                          False   False               False      False   
2                          False   False               False      False   
3                          False   False               False      False   
4                          False   False               False      False   

   КлиентПолПредставление  ДатаЗавершенияМедосмотра  ТипМедосмотра  \
0                   False                     False          False   
1                   False                     False          False   
2                   False                     False          False   
3                   False                     False          False   
4                   False                     False          False   

   ТипМедосмотраПредставление  ПсихОсвидетельствование  Контрагент  ...  \
0                       False                    False       False  ...   
1                       False                    False       False  ...   
2                       False                    False       False  ...   
3                       False                    False       False  ...   
4                       False                    False       False  ...   

   Оториноларингология3_Аудиометрия1_Значение  \
0                                        True   
1                                        True   
2                                        True   
3                                        True   
4                                        True   

   Оториноларингология3_Аудиометрия1_ЗначениеПредставление  \
0                                               True         
1                                               True         
2                                               True         
3                                               True         
4                                               True         

   Оториноларингология1_ЖалобыДополнение_Значение  \
0                                            True   
1                                            True   
2                                            True   
3                                            True   
4                                            True   

   Оториноларингология1_ЖалобыДополнение_ЗначениеПредставление  \
0                                               True             
1                                               True             
2                                               True             
3                                               True             
4                                               True             

   Стоматология2_ВП2_Значение  Стоматология2_ВП2_ЗначениеПредставление  \
0                        True                                     True   
1                        True                                     True   
2                        True                                     True   
3                        True                                     True   
4                        True                                     True   

   Терапия2_ЗаключениеФОГК_Значение  \
0                              True   
1                              True   
2                              True   
3                              True   
4                              True   

   Терапия2_ЗаключениеФОГК_ЗначениеПредставление  \
0                                           True   
1                                           True   
2                                           True   
3                                           True   
4                                           True   

   Офтальмология1_ДвигательныйАппаратГлазДополнение_Значение  \
0                                               True           
1                                               True           
2                                               True           
3                                              

In [18]:
#Сколько пустых ячеек?
data.isna().sum()

ДокументПрохождениеМедосмотра                                                0
Клиент                                                                       0
КлиентДатаРождения                                                           0
КлиентПол                                                                    0
КлиентПолПредставление                                                       0
                                                                          ... 
Стоматология2_ВП2_ЗначениеПредставление                                   1981
Терапия2_ЗаключениеФОГК_Значение                                          1981
Терапия2_ЗаключениеФОГК_ЗначениеПредставление                             1981
Офтальмология1_ДвигательныйАппаратГлазДополнение_Значение                 1981
Офтальмология1_ДвигательныйАппаратГлазДополнение_ЗначениеПредставление    1981
Length: 4587, dtype: int64

Вообщем и целом видим какое-то количество пропусков есть.

In [20]:
#Доля пропусков от общего количества элементов
data.isna().sum() / data.shape[0]

ДокументПрохождениеМедосмотра                                             0.000000
Клиент                                                                    0.000000
КлиентДатаРождения                                                        0.000000
КлиентПол                                                                 0.000000
КлиентПолПредставление                                                    0.000000
                                                                            ...   
Стоматология2_ВП2_ЗначениеПредставление                                   0.999495
Терапия2_ЗаключениеФОГК_Значение                                          0.999495
Терапия2_ЗаключениеФОГК_ЗначениеПредставление                             0.999495
Офтальмология1_ДвигательныйАппаратГлазДополнение_Значение                 0.999495
Офтальмология1_ДвигательныйАппаратГлазДополнение_ЗначениеПредставление    0.999495
Length: 4587, dtype: float64

In [21]:
# Общая сумма пропусков
data.isna().sum().sum()

8106978

In [22]:
# Общий процент пропусков
proc = data.isna().sum().sum() / data.size
print('{:.1%}'.format(proc))

89.2%


Исходя из этих данных получается что пропущено 90% строк?

Параметр include=['O'] метода .describe() выведет статистику для строковых данных:

*   count - количество заполненных элементов
*   unique - количество уникальных элементов
*   top - самый распространенный элемент
*   freq - сколько раз встречается самый распространенный элемент


In [24]:
# Вывод статистики для строковых данных.
data.describe(include=['O'])

ДокументПрохождениеМедосмотра  \
count                                   1982   
unique                                  1915   
top     e334cb0e-55a6-11ed-8108-0cc47aab8067   
freq                                       2   

                                      Клиент КлиентПол КлиентПолПредставление  \
count                                   1982      1982                   1982   
unique                                  1911         2                      2   
top     116fe6a1-455f-11ed-8107-0cc47aab8067   Мужской                Мужской   
freq                                       3      1451                   1451   

        ТипМедосмотра ТипМедосмотраПредставление  \
count            1982                       1982   
unique              2                          2   
top     Периодический              Периодический   
freq             1981                       1981   

                                  Контрагент         Цех Профессия  \
count                                   1982         802      1982   
unique                                   177         183       690   
top     173d6ba7-12fe-11e7-80d3-708bcd7f83cd  Без отдела  Водитель   
freq                                     162          63        55   

       ЗаключениеМК  ... Оториноларингология3_Аудиометрия1_Значение  \
count          1575  ...                                          1   
unique            9  ...                                          1   
top           Годен  ...       2934f2d1-8e1f-11ec-80fe-0cc47aab8067   
freq           1434  ...                                          1   

       Оториноларингология3_Аудиометрия1_ЗначениеПредставление  \
count                                                   1        
unique                                                  1        
top                              не подлежит обследованию        
freq                                                    1        

       Оториноларингология1_ЖалобыДополнение_Значение  \
count                                               1   
unique                                              1   
top                                   на ком в горле.   
freq                                                1   

       Оториноларингология1_ЖалобыДополнение_ЗначениеПредставление  \
count                                                   1            
unique                                                  1            
top                                       на ком в горле.            
freq                                                    1            

       Стоматология2_ВП2_Значение Стоматология2_ВП2_ЗначениеПредставление  \
count                           1                                       1   
unique                          1                                       1   
top                             C                                       C   
freq                            1                                       1   

                         Терапия2_ЗаключениеФОГК_Значение  \
count                                                   1   
unique                                                  1   
top     Рекомендации: лекарственные препараты принимат...   
freq                                                    1   

            Терапия2_ЗаключениеФОГК_ЗначениеПредставление  \
count                                                   1   
unique                                                  1   
top     Рекомендации: лекарственные препараты принимат...   
freq                                                    1   

       Офтальмология1_ДвигательныйАппаратГлазДополнение_Значение  \
count                                                   1          
unique                                                  1          
top           девиация левого глаза к носу до 25 градусов          
freq                                                    1          

       Офтальмология1_ДвигательныйАппаратГлазДополнение_ЗначениеПредставление  
count        

**Выводы сделанные на основе анализа датасета**

*   Имеет 1982 строки и 4587 ячеек .
*   Общие данные после команды data.inf()
*   Почему-то общий процент пропусков 90%.
*   Есть уникальные идентификаторы. Они не нужны.
*   Есть закономерности, такие как пол влияет на прохождение врачей.
*   Нас иньересуют только 3 класса-**годен,ограниченно годен,негоден**.
*   Если не прошел психОсведетельствование это **Негоден**.


# Создание выборок

In [43]:
# Создадим копию, так как будем редактировать таблицу.
data_backup = data.copy()
data.head()

ДокументПрохождениеМедосмотра                                Клиент  \
0  7f41ca7e-7dac-11ec-80fc-0cc47aab8067  c91e3407-b81f-11e9-80cf-0cc47aab8067   
1  2ee1f9ac-83dd-11ec-80fc-0cc47aab8067  1a1a5fea-8b7f-11eb-80e8-0cc47aab8067   
2  9cbb191e-8e24-11ec-80fe-0cc47aab8067  7fe324e0-fa51-11eb-80f6-0cc47aab8067   
3  b8a33910-8eec-11ec-80fe-0cc47aab8067  98f9d761-8d65-11ec-80fe-0cc47aab8067   
4  0bb0d181-938c-11ec-80fe-0cc47aab8067  009aac7d-7bcf-11eb-80e8-0cc47aab8067   

  КлиентДатаРождения КлиентПол КлиентПолПредставление  \
0         1976-10-24   Женский                Женский   
1         1987-11-18   Мужской                Мужской   
2         1985-09-08   Мужской                Мужской   
3         1988-04-25   Мужской                Мужской   
4         1984-04-17   Мужской                Мужской   

  ДатаЗавершенияМедосмотра  ТипМедосмотра ТипМедосмотраПредставление  \
0               2022-01-25  Периодический              Периодический   
1               2022-02-02  Периодический              Периодический   
2               2022-02-15  Периодический              Периодический   
3               2022-02-16  Периодический              Периодический   
4               2022-02-22  Периодический              Периодический   

   ПсихОсвидетельствование                            Контрагент  ...  \
0                    False  23c173a7-30d9-11e7-80d8-708bcd7f83cd  ...   
1                    False  a6d559b0-7835-11eb-80e8-0cc47aab8067  ...   
2                    False  a6d559b0-7835-11eb-80e8-0cc47aab8067  ...   
3                    False  a6d559b0-7835-11eb-80e8-0cc47aab8067  ...   
4                    False  a6d559b0-7835-11eb-80e8-0cc47aab8067  ...   

  Оториноларингология3_Аудиометрия1_Значение  \
0                                        NaN   
1                                        NaN   
2                                        NaN   
3                                        NaN   
4                                        NaN   

  Оториноларингология3_Аудиометрия1_ЗначениеПредставление  \
0                                                NaN        
1                                                NaN        
2                                                NaN        
3                                                NaN        
4                                                NaN        

  Оториноларингология1_ЖалобыДополнение_Значение  \
0                                            NaN   
1                                            NaN   
2                                            NaN   
3                                            NaN   
4                                            NaN   

  Оториноларингология1_ЖалобыДополнение_ЗначениеПредставление  \
0                                                NaN            
1                                                NaN            
2                                                NaN            
3                                                NaN            
4                                                NaN            

  Стоматология2_ВП2_Значение Стоматология2_ВП2_ЗначениеПредставление  \
0                        NaN                                     NaN   
1                        NaN                                     NaN   
2                        NaN                                     NaN   
3                        NaN                                     NaN   
4                        NaN                                     NaN   

  Терапия2_ЗаключениеФОГК_Значение  \
0                              NaN   
1                              NaN   
2                              NaN   
3                              NaN   
4                              NaN   

  Терапия2_ЗаключениеФОГК_ЗначениеПредставление  \
0                                           NaN   
1                                           NaN   
2                                           NaN   
3                                           NaN   
4                              

In [44]:
#Восстановление датасета.
data = data_backup.copy()
data.head()

ДокументПрохождениеМедосмотра                                Клиент  \
0  7f41ca7e-7dac-11ec-80fc-0cc47aab8067  c91e3407-b81f-11e9-80cf-0cc47aab8067   
1  2ee1f9ac-83dd-11ec-80fc-0cc47aab8067  1a1a5fea-8b7f-11eb-80e8-0cc47aab8067   
2  9cbb191e-8e24-11ec-80fe-0cc47aab8067  7fe324e0-fa51-11eb-80f6-0cc47aab8067   
3  b8a33910-8eec-11ec-80fe-0cc47aab8067  98f9d761-8d65-11ec-80fe-0cc47aab8067   
4  0bb0d181-938c-11ec-80fe-0cc47aab8067  009aac7d-7bcf-11eb-80e8-0cc47aab8067   

  КлиентДатаРождения КлиентПол КлиентПолПредставление  \
0         1976-10-24   Женский                Женский   
1         1987-11-18   Мужской                Мужской   
2         1985-09-08   Мужской                Мужской   
3         1988-04-25   Мужской                Мужской   
4         1984-04-17   Мужской                Мужской   

  ДатаЗавершенияМедосмотра  ТипМедосмотра ТипМедосмотраПредставление  \
0               2022-01-25  Периодический              Периодический   
1               2022-02-02  Периодический              Периодический   
2               2022-02-15  Периодический              Периодический   
3               2022-02-16  Периодический              Периодический   
4               2022-02-22  Периодический              Периодический   

   ПсихОсвидетельствование                            Контрагент  ...  \
0                    False  23c173a7-30d9-11e7-80d8-708bcd7f83cd  ...   
1                    False  a6d559b0-7835-11eb-80e8-0cc47aab8067  ...   
2                    False  a6d559b0-7835-11eb-80e8-0cc47aab8067  ...   
3                    False  a6d559b0-7835-11eb-80e8-0cc47aab8067  ...   
4                    False  a6d559b0-7835-11eb-80e8-0cc47aab8067  ...   

  Оториноларингология3_Аудиометрия1_Значение  \
0                                        NaN   
1                                        NaN   
2                                        NaN   
3                                        NaN   
4                                        NaN   

  Оториноларингология3_Аудиометрия1_ЗначениеПредставление  \
0                                                NaN        
1                                                NaN        
2                                                NaN        
3                                                NaN        
4                                                NaN        

  Оториноларингология1_ЖалобыДополнение_Значение  \
0                                            NaN   
1                                            NaN   
2                                            NaN   
3                                            NaN   
4                                            NaN   

  Оториноларингология1_ЖалобыДополнение_ЗначениеПредставление  \
0                                                NaN            
1                                                NaN            
2                                                NaN            
3                                                NaN            
4                                                NaN            

  Стоматология2_ВП2_Значение Стоматология2_ВП2_ЗначениеПредставление  \
0                        NaN                                     NaN   
1                        NaN                                     NaN   
2                        NaN                                     NaN   
3                        NaN                                     NaN   
4                        NaN                                     NaN   

  Терапия2_ЗаключениеФОГК_Значение  \
0                              NaN   
1                              NaN   
2                              NaN   
3                              NaN   
4                              NaN   

  Терапия2_ЗаключениеФОГК_ЗначениеПредставление  \
0                                           NaN   
1                                           NaN   
2                                           NaN   
3                                           NaN   
4                              

**Удалю ненужные столбцы.**

In [45]:
# Указать список столбцов, которые нужно удалить
columns_to_drop = ['ДокументПрохождениеМедосмотра', 'Клиент', 'Контрагент']

# Удалить указанные столбцы из датасета
data.drop(columns=columns_to_drop, inplace=True)
data.head()

КлиентДатаРождения КлиентПол КлиентПолПредставление  \
0         1976-10-24   Женский                Женский   
1         1987-11-18   Мужской                Мужской   
2         1985-09-08   Мужской                Мужской   
3         1988-04-25   Мужской                Мужской   
4         1984-04-17   Мужской                Мужской   

  ДатаЗавершенияМедосмотра  ТипМедосмотра ТипМедосмотраПредставление  \
0               2022-01-25  Периодический              Периодический   
1               2022-02-02  Периодический              Периодический   
2               2022-02-15  Периодический              Периодический   
3               2022-02-16  Периодический              Периодический   
4               2022-02-22  Периодический              Периодический   

   ПсихОсвидетельствование                                                Цех  \
0                    False                                                NaN   
1                    False                                                NaN   
2                    False                                                NaN   
3                    False   Отдел техники и технологии строительства скважин   
4                    False  Отдел технологий текущего и капитального ремон...   

                                   Профессия            ЗаключениеМК  ...  \
0  Ведущий специалист по организации обуения                   Годен  ...   
1                         Ведущий специалист                   Годен  ...   
2                           Главный менеджер  ГоденСКоррекциейЗрения  ...   
3                         Главный специалист                   Годен  ...   
4                         Старший специалист  ГоденСКоррекциейЗрения  ...   

  Оториноларингология3_Аудиометрия1_Значение  \
0                                        NaN   
1                                        NaN   
2                                        NaN   
3                                        NaN   
4                                        NaN   

  Оториноларингология3_Аудиометрия1_ЗначениеПредставление  \
0                                                NaN        
1                                                NaN        
2                                                NaN        
3                                                NaN        
4                                                NaN        

  Оториноларингология1_ЖалобыДополнение_Значение  \
0                                            NaN   
1                                            NaN   
2                                            NaN   
3                                            NaN   
4                                            NaN   

  Оториноларингология1_ЖалобыДополнение_ЗначениеПредставление  \
0                                                NaN            
1                                                NaN            
2                                                NaN            
3                                                NaN            
4                                                NaN            

  Стоматология2_ВП2_Значение Стоматология2_ВП2_ЗначениеПредставление  \
0                        NaN                                     NaN   
1                        NaN                                     NaN   
2                        NaN                                     NaN   
3                        NaN                                     NaN   
4                        NaN                                     NaN   

  Терапия2_ЗаключениеФОГК_Значение  \
0                              NaN   
1                              NaN   
2                              NaN   
3                              NaN   
4                              NaN   

  Терапия2_ЗаключениеФОГК_ЗначениеПредставление  \
0                                           NaN   
1                                           NaN   
2                                           NaN   
3                                           NaN   

**Переводим в NumPy-массив**

In [46]:
# Получение данных из таблицы и их запись в массив np_data
np_data = data.values 

In [48]:
# Проверяем данные
print(np_data.shape)    # Вывод формы массива np_data
print(np_data[0])       # Вывод 0-ой строки массива

(1982, 4584)
[Timestamp('1976-10-24 00:00:00') 'Женский' 'Женский' ... nan nan nan]


In [49]:
# Вывод первых 5 элементов массива np_data
for i in range(5):
    print(np_data[i])

[Timestamp('1976-10-24 00:00:00') 'Женский' 'Женский' ... nan nan nan]
[Timestamp('1987-11-18 00:00:00') 'Мужской' 'Мужской' ... nan nan nan]
[Timestamp('1985-09-08 00:00:00') 'Мужской' 'Мужской' ... nan nan nan]
[Timestamp('1988-04-25 00:00:00') 'Мужской' 'Мужской' ... nan nan nan]
[Timestamp('1984-04-17 00:00:00') 'Мужской' 'Мужской' ... nan nan nan]
